### PR TITLE
Skip default adapters for translation layer clients

### PR DESCRIPTION
**Pull request by abhishkk**
RFC: https://github.com/Microsoft/vstest-docs/pull/122 

Validations done:
1. When SkipDefaultAdapters is set true, tests are running as expected.
2. When SkipDefaultAdapters is set false, tests are running as expected.
3. When SkipDefaultAdapters is not set, tests are running as expected.
4. cpp tests, CUIT tests running fine.
5. LUT and IDE scenarios working fine.

### CODE REVIEW

File: src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyExecutionManager.cs [hunk 1]
```diff
@@ -257,7 +260,7 @@ private void LogMessage(TestMessageLevel testMessageLevel, string message)
 
         private void InitializeExtensions(IEnumerable<string> sources)
         {
-            var extensions = TestPluginCache.Instance.GetExtensionPaths(TestPlatformConstants.TestAdapterEndsWithPattern);
+            var extensions = TestPluginCache.Instance.GetExtensionPaths(TestPlatformConstants.TestAdapterEndsWithPattern, skipDefaultAdapters);
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyExecutionManager.cs]: nit: use this. with private variables

File: src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyDiscoveryManager.cs [hunk 2]
```diff
@@ -175,7 +178,7 @@ public void HandleLogMessage(TestMessageLevel level, string message)
 
         private void InitializeExtensions(IEnumerable<string> sources)
         {
-            var extensions = TestPluginCache.Instance.GetExtensionPaths(TestPlatformConstants.TestAdapterEndsWithPattern);
+            var extensions = TestPluginCache.Instance.GetExtensionPaths(TestPlatformConstants.TestAdapterEndsWithPattern, skipDefaultAdapters);
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyDiscoveryManager.cs]: nit: use this.

File: src/Microsoft.TestPlatform.ObjectModel/Client/Interfaces/ITestPlatform.cs [hunk 3]
```diff
@@ -23,20 +23,23 @@ public interface ITestPlatform : IDisposable
         /// </summary>
         void ClearExtensions();
 
+
         /// <summary>
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.ObjectModel/Client/Interfaces/ITestPlatform.cs]: nit: Extra line above

**Review by mayankbansal018 — APPROVED** [review]
:shipit:

File: src/Microsoft.TestPlatform.ObjectModel/Client/TestPlatformOptions.cs [hunk 4]
```diff
@@ -34,5 +34,11 @@ public class TestPlatformOptions
         /// </summary>
         [DataMember]
         public bool CollectMetrics { get; set; }
+
+        /// <summary>
+        ///  Gets or sets whether default adapters should be skipped or not.
+        /// </summary>
+        [DataMember]
+        public bool SkipDefaultAdapters { get; set; }
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.ObjectModel/Client/TestPlatformOptions.cs]: >SkipDefaultAdapters [](start = 20, length = 19)

Nit: rename to SkipDefaultExtensions?

Comment (abhishkk) [reply src/Microsoft.TestPlatform.ObjectModel/Client/TestPlatformOptions.cs]: We are skipping default adapters only and not default extensions. So as per discussion with Shiva and Pratap, made it SkipDefaultAdapters

**Review by singhsarab — APPROVED** [review]

**Pull request merged**